### PR TITLE
feat: support React 18

### LINKED ISSUE #163: Support React v18
Do you plan to upgrade to React 18 to avoid npm dependency conflict?

### LINKED ISSUE #166: Update typedef to support React 18 Typedef changes
As per a PR that commited for React 18 typedefs, there are type warnings when using these typedefs due to missing children props.

>Due to changes in the React18 typedefs https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56210
We plan to remove implicit children from @types/react.

### PR DESCRIPTION
**Pull request by andipaetzold**
React 18 doesn't have any relevant breaking changes.

closes #163
closes #166

### CODE REVIEW

**Comment by bacali95** [comment]
Hello @andipaetzold, thanks for adding support for React 18, just wanted to mention that the types of React do not use `React.PropsWithChildren` type by default for class and functional components see https://github.com/DefinitelyTyped/DefinitelyTyped/commit/55dc209ceb6dbcd59c4c68cc8dfb77faadd9de12

So can you please alter the `HelmetProvider` type to have children as a prop?
```ts
...
export class HelmetProvider extends React.Component<React.PropsWithChildren<ProviderProps>> {
  static canUseDOM: boolean;
}
...
```
Thanks a lot :tulip:

**Comment by andipaetzold** [comment]
@bacali95 Good catch, `@types/react@18` was released after I created the PR, so I missed the changes.

I am gonna update the PR to ensure it is compatible with both `@types/react@17` and `@types/react@18`.

**Comment by andipaetzold** [comment]
@bacali95 I just updated the PR

File: index.d.ts [hunk 1]
```diff
@@ -41,9 +41,8 @@ declare module 'react-helmet-async' {
     titleTemplate?: string;
   }
 
-  export class Helmet extends React.Component<HelmetProps> {
-  }
-  
+  export class Helmet extends React.Component<HelmetProps> {}
```
Comment (bacali95) [inline index.d.ts:44]: I think this one also needs `PropsWithChildren` type

**Comment by bacali95** [comment]
Hey @andipaetzold thanks for the quick fix :tulip: 

Hey @staylor can please review/approve this PR :tulip:

File: index.d.ts (hunk 1 above)
Comment (andipaetzold) [reply index.d.ts:44]: I am super sure I added it there 😆

Comment (bacali95) [reply index.d.ts]: It could be you didn't stage it :stuck_out_tongue:

**Comment by staylor** [comment]
this needs to be rebased on your side, then I can merge it

**Comment by andipaetzold** [comment]
@staylor done

**Comment by dquoctri** [comment]
I have a workaround to fix this issue by myself. I'm waiting for this PR.
Thank you all!

export interface HelmetProps {
    ...
    children?: React.ReactNode;
  }

interface ProviderProps {
    context?: {};
    children?: React.ReactNode;
  }

**Comment by kimbaudi** [comment]
Thanks for this PR. Looking forward to the next release. 😄

**Comment by andipaetzold** [comment]
Thank you 🚀

**Pull request merged**